runtime/cudaq/platform/default/rest/helpers/oqc/OQCServerHelp.cpp

```diff
@@ -0,0 +1,255 @@
+  /// retrieval request.
+  bool jobIsDone(ServerMessage &getJobResponse) override;
+
+  /// @brief Processes the server's response to a job retrieval request and
+  /// maps the results back to sample results.
+  cudaq::sample_result processResults(ServerMessage &postJobResponse) override;
+};
+
+// Initialize the OQC server helper with a given backend configuration
+void OQCServerHelper::initialize(BackendConfig config) {
+  cudaq::info("Initializing OQC Backend.");
+  // Move the passed config into the member variable backendConfig
+  backendConfig = std::move(config);
+  // Set the necessary configuration variables for the OQC API
+  backendConfig["url"] = config.find("url") != config.end()
+                             ? config["url"]
+                             : "http://127.0.0.1";
+  backendConfig["version"] = "v0.3";
+  backendConfig["user_agent"] = "cudaq/0.3.0";
+  backendConfig["target"] =
+      config.find("qpu") != config.end() ? config["qpu"] : "simulator";
+  backendConfig["qubits"] = 8;
+  // Retrieve the API key from the environment variables
+  backendConfig["email"] = getEnvVar("OQC_EMAIL");
+  backendConfig["password"] = getEnvVar("OQC_PASSWORD");
+  // Construct the API job path
+  backendConfig["job_path"] =
+      backendConfig["url"] + "/tasks";
+}
+
+// Retrieve an environment variable
+std::string OQCServerHelper::getEnvVar(const std::string &key) const {
+  // Get the environment variable
+  const char *env_var = std::getenv(key.c_str());
+  // If the variable is not set, throw an exception
+  if (env_var == nullptr) {
+    throw std::runtime_error(key + " environment variable is not set.");
+  }
+  // Return the variable as a string
+  return std::string(env_var);
+}
+
+// Check if a key exists in the backend configuration
+bool OQCServerHelper::keyExists(const std::string &key) const {
+  return backendConfig.find(key) != backendConfig.end();
+}
+
+std::vector<std::string> OQCServerHelper::getJobID(int n){
+    RestHeaders headers = this -> getHeaders();
+    return client.get(backendConfig.at("job_path")+"/?" + std::to_string(n), "", headers );
+}
+
+std::string OQCServerHelper::makeConfig(int shots){
+    return "{\"$type\": \"<class 'scc.compiler.config.CompilerConfig'>\", \"$data\": {\"repeats\": "+std::to_string(shots)+", \"repetition_period\": null, \"results_format\": {\"$type\": \"<class 'scc.compiler.config.QuantumResultsFormat'>\", \"$data\": {\"format\": {\"$type\": \"<enum 'scc.compiler.config.InlineResultsProcessing'>\", \"$value\": 1}, \"transforms\": {\"$type\": \"<enum 'scc.compiler.config.ResultsFormatting'>\", \"$value\": 3}}}, \"metrics\": {\"$type\": \"<enum 'scc.compiler.config.MetricsType'>\", \"$value\": 6}, \"active_calibrations\": [], \"optimizations\": {\"$type\": \"<class 'scc.compiler.config.Tket'>\", \"$data\": {\"tket_optimizations\": {\"$type\": \"<enum 'scc.compiler.config.TketOptimizations'>\", \"$value\": 30}}}}}";
```

Comment:
Here and else where:

Define a `std::map<std::string, std::string>` fill with entries ` = { {key, value}, {key, value} }` then just `nlohmann::json_v3_11_1::json(map)`

I'll push some changes with similar in.